.github/workflows/images.yml

```diff
@@ -0,0 +1,50 @@
+name: Image Push
+on:
+  push:
+    branches:
+      - develop
+      - 'release/**'
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+**'
+
+jobs:
+  image-build-push:
+    runs-on: ubuntu-latest
+    permissions:
+        contents: read
+        id-token: write
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+          fetch-depth: 0
+      - run: |
+          # BUG: HEAD tag is fetched as lightweight instead of annotated
+          # https://github.com/actions/checkout/issues/290
+          if [ "${{ github.ref_type }}" == "tag" ]; then
+            git fetch -f origin ${{ github.ref }}:${{ github.ref }}
+          fi
+
+      - uses: DeterminateSystems/nix-installer-action@v14
+        with:
+            github-token: '' 
+
+      - uses: DeterminateSystems/magic-nix-cache-action@v8
+
+      - name: Pre-populate nix-shell
+        run: |
+            export NIX_PATH=nixpkgs=$(jq '.nixpkgs.url' nix/sources.json -r)
+            echo "NIX_PATH=$NIX_PATH" >> $GITHUB_ENV
+            nix-shell --run "echo" shell.nix
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and push the release images
+        run: |
+          TAG=$(nix-shell --run "cat charts/Chart.yaml | yq .appVersion")
+          nix-shell --run "./scripts/release.sh --image upgrade.job --tag v${TAG}"
```

Comment:
This is okay for regular pushes to the branch, but for tag pushes, the tag should be the pushed tag.

Reply:
Yes ! I missed that. Let me update the workflow.